test/ibmq/test_ibmq_provider.py

```diff
@@ -64,6 +65,14 @@ def test_remote_backends_exist_simulator(self):
         remotes = self.provider.backends(simulator=True)
         self.assertTrue(remotes)
 
+    def test_remote_backends_instantiate_simulators(self):
+        """Test if remote backends that are simulators
+        get instantiated as an IBMQSimulator class."""
+        remotes = self.provider.backends(simulator=True)
+
+        for backend in remotes:
+            self.assertIsInstance(backend, IBMQSimulator)
```

Comment:
Instead of making the assertions in a loop directly, can you make use of [`subTest()`](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests)? The main practical difference is :

> Without using a subtest, execution would stop after the first failure, and the error would be less easy to diagnose because the value of i wouldn’t be displayed:

In general, it is a way of "splitting" a test into several subtests that are run "_independently_" - there should not be much difference in this particular test, but it would help in an hypothetical case where one particular backend fails to be of the required class: with subtests, we could still be able to assert over the rest of the backends, isolating that single failure.

Reply:
That's very interesting. I wasn't aware that future tests would fail in this case. This will be very helpful for future testing. Thank you.